src/affinity/binders.c

```diff
@@ -91,9 +91,15 @@ int INTERNAL qt_affinity_compact(int num_workers,
 void INTERNAL qt_affinity_init(qthread_shepherd_id_t *nbshepherds,
                                qthread_worker_id_t   *nbworkers,
                                size_t                *hw_par)
-{                           
-  hwloc_topology_init(&topology);
-  hwloc_topology_load(topology);
+{
+#ifdef HWLOC_GET_TOPOLOGY_FUNCTION
+  extern void * HWLOC_GET_TOPOLOGY_FUNCTION;
+  topology = (hwloc_topology_t) HWLOC_GET_TOPOLOGY_FUNCTION;
+#endif
+  if (topology == NULL) {
```

Comment:
What about removing this conditional

Reply:
I see your point.

Reply:
@jhh67 Could you add a comment into the code that the lack of a tear_down routine makes qt_affinity_init skip topology initialization in case of qthreads reinitialization? If we update binders at some point we'd likely need to add a tear_down routine.

Reply:
Done.